frontends/web/src/containers/ModelPage.js

```diff
@@ -31,99 +31,123 @@ import FloresGrid from "../components/FloresComponents/FloresGrid";
+            <Modal.Body>
+              {tag.dataset_longdesc}
+              <br />
+              <br />
+              {tag.dataset_source_url && tag.dataset_source_url !== "" ? (
+                <Button href={tag.dataset_source_url}>
+                  <i className="fas fa-newspaper"></i> Read Paper
+                </Button>
+              ) : (
+                ""
+              )}
+            </Modal.Body>
+          </Modal>
+        ) : (
+          ""
+        )}
+        <tr style={{ border: `none` }}>
+          <td className="text-right pr-4  text-nowrap">
+            <div className="d-flex align-items-center justify-content-between">
+              {tag.root ? (
+                <span
+                  onClick={() => setShowModal(!showModal)}
+                  className="btn-link dataset-link"
+                >
+                  {expanded ? <b>{tag.dataset_name}</b> : tag.dataset_name}
+                </span>
+              ) : (
+                <span className="d-flex align-items-center">
+                  {tag.tag}&nbsp;
+                </span>
+              )}
+              {clickable ? (
+                <div onClick={() => (clickable ? setExpanded(!expanded) : "")}>
+                  <ChevronExpandButton
+                    expanded={expanded}
+                    containerClassName={"start-100"}
+                  />
+                </div>
               ) : (
-                parseFloat(score.accuracy).toFixed(2)
+                ""
               )}
-            </span>
+            </div>
+          </td>
+          <td className="text-right " key={`score-${tag.tag}-overall`}>
+            <span>{parseFloat(tag.perf).toFixed(2)}</span>
           </td>
         </tr>
         {expanded &&
-          perf_by_tag.map((perf_and_tag, index) => {
-            return (
-              <tr key={index} style={{ border: `none` }}>
-                <td className="text-right pr-4  text-nowrap">
-                  <div className="d-flex justify-content-end align-items-center">
-                    <span className="d-flex align-items-center">
-                      {perf_and_tag.tag}&nbsp;
-                    </span>
-                  </div>
-                </td>
-                <td
-                  className="text-right "
-                  key={`score-${score.dataset_name}-${perf_and_tag.tag}-overall`}
-                >
-                  <span>{parseFloat(perf_and_tag.perf).toFixed(2)}</span>
-                </td>
-              </tr>
-            );
+          tag.subTags &&
+          tag.subTags.map((new_tag) => {
+            return <TreeRow tag={new_tag} />;
           })}
       </tbody>
     </Table>
   );
 };
 
+const ScoreRow = ({ score }) => {
+  const perf_by_tag =
+    score.metadata_json &&
+    JSON.parse(score.metadata_json).hasOwnProperty("perf_by_tag")
+      ? JSON.parse(score.metadata_json)["perf_by_tag"]
+      : [];
+
+  const tag_hierarchy = JSON.parse(score.dataset_tag_hierarchy);
```

Comment:
This part reads in the `tag_hierarchy` the user suggested, and modifies the hierarchy in place with the `perf` values (basically just doing bfs on the tag hierarchy and modifying the object in place with values from `score`)